lib/nx_gzlib.c

```diff
@@ -0,0 +1,105 @@
+/* nx_gzlib.c -- implement gz* functions
+ *
+ * NX-GZIP compression accelerator user library
+ * implementing zlib compression library interfaces
+ *
+ * Copyright (C) IBM Corporation, 2022
+ *
+ * Licenses for GPLv2 and Apache v2.0:
+ *
+ * GPLv2:
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ *
+ * Apache v2.0:
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "nx_zlib.h"
+
+#define DEF_MEM_LEVEL 8
+
+gzFile nx_gzopen_(const char* path, int fd, const char *mode);
+
+typedef struct gz_state {
+	gzFile gz;
+	FILE *fd;
+	int err;
+	z_stream strm;
+} *gz_statep;
+
+int nx_gzclose(gzFile file)
+{
+	gz_statep state = (gz_statep) file;
+	z_streamp strm = &(state->strm);
+	int ret;
+
+	if(file == NULL) return Z_STREAM_ERROR;
+
+	ret = nx_inflateEnd(strm);
+
+	return ret;
+}
+
+gzFile nx_gzdopen(int fd, const char *mode)
+{
+	return nx_gzopen_(NULL, fd, mode);
+}
+
+gzFile nx_gzopen(const char* path, const char *mode)
+{
+	return nx_gzopen_(path, 0, mode);
+}
+
+gzFile nx_gzopen_(const char* path, int fd, const char *mode)
+{
+	gz_statep state;
+	gzFile file;
+	int err;
+	void* strm;
+
+	state = malloc(sizeof(struct gz_state));
+
+	if(path == NULL)
+		state->fd = fdopen(fd, mode);
+	else
+		state->fd = fopen(path, mode);
+	if(state->fd == NULL){
+		free(state);
+		return NULL;
+	}
+
+	strm = &(state->strm);
+	memset(strm, 0, sizeof(z_stream));
+	if(strchr(mode, 'w')){
+		err = nx_deflateInit2_(&(state->strm), Z_DEFAULT_COMPRESSION, Z_DEFLATED, 31, DEF_MEM_LEVEL, Z_DEFAULT_STRATEGY, ZLIB_VERSION, (int)sizeof(z_stream));
```

Comment:
This is a very long line.
Could you limit it to 80 characters, please?

Reply:
@rzinsly I added `nx_deflateInit2` as part of an upcoming PR, but I split the commit into #139 in case you want to use it here. It will help make this line shorter.